src/redprl/pretty.sml

```diff
@@ -83,10 +82,33 @@ struct
        [] => empty
      | _ => m
 
+  val ppIntInf = text o IntInf.toString
+
+  val rec ppParam =
+    fn P.VAR x => ppSym x
+     | P.APP (P.LCONST i) => ppIntInf i
+     | P.APP (P.LABOVE (p, 1)) => Atomic.braces @@ hsep [text "lsucc", ppParam p]
+     | P.APP (P.LABOVE (p, i)) => Atomic.braces @@ hsep [text "labove", ppParam p, ppIntInf i]
+     | P.APP (P.LMAX ps) => Atomic.braces @@ hsep @@ text "lmax" :: List.map ppParam ps
+     | P.APP P.DIM0 => text "0"
+     | P.APP P.DIM1 => text "1"
+
+  val ppParamWithSort =
+    fn (p, NONE) => ppParam p
+     | (P.VAR x, SOME s) => Atomic.braces @@ hsep [text @@ S.paramSortToString s, ppSym x]
```

Comment:
I am not sure if providing the sort here is needed, but it should be at least safe.

Reply:
This will in practice always be `SOME`; the reason that it is `option` at all is that at parsetime, we don't necessarily know the sort, and these are filled in during elaboration.

Reply:
Since the printer should match the parser, I suppose you saying that the sorts can be omitted for the variables.